pkg/users/local.js

```diff
@@ -873,6 +903,44 @@ PageAccount.prototype = {
            });
     },
 
+    check_tlog_shell: function() {
+        cockpit.file("/usr/bin/tlog-rec-session", { binary: true }).read().done( function(content, tag) {
+            if (content != null) {
+                $("#session_recoding_tr").show();
+            } else {
+                $("#session_recoding_tr").hide();
+            }
+        }).fail( function(error) {
+            console.log(error);
+        });
+    },
+
+    get_recorded: function() {
+        var self = this;
+        function get_shell(content) {
+            var accounts = parse_passwd_content(content);
+
+            for (var i = 0; i < accounts.length; i++) {
+                if (accounts[i]["name"] !== self.account_id)
+                    continue;
+
+                self.account = accounts[i];
+                if (self.account.shell === "/usr/bin/tlog-rec-session") {
+                    self.account.recorded = true;
+                } else {
+                    self.account.recorded = false;
+                }
```

Comment:
Style nitpick: `self.account.recorded = shell === SESSION_SHELL` (this will become part of the `.filter()` function).

Reply:
This is actually abusing `.filter()` a bit, as you modify things inline and don't use the return value. If you want to do it this way, just use `.forEach()` instead of `.filter()` to be more explicit.